File: apps/client/src/types/project.ts
// ── Project ───────────────────────────────────────────
export interface Project {
  id: string
  title: string
  genre?: string
  logline?: string
  premise?: string
  world_overview?: string
  story_core?: Record<string, any>
  status: 'drafting' | 'writing' | 'completed'
  target_words?: number
  cover_url?: string
  created_at: string
  updated_at?: string
  /** JSONB 杂物字段；已知键：positioning / writing_config / opening_contract / consistency_issues */
  extra?: Record<string, any>
}

File: apps/backend/app/schemas/project.py
from pydantic import BaseModel
from typing import Optional, Any
from datetime import datetime
import uuid


class ProjectCreate(BaseModel):
    title: str
    genre: Optional[str] = None
    logline: Optional[str] = None
    premise: Optional[str] = None
    world_overview: Optional[str] = None
    story_core: Optional[dict] = {}
    target_words: Optional[int] = None


class ProjectUpdate(BaseModel):
    title: Optional[str] = None
    genre: Optional[str] = None
    logline: Optional[str] = None
    premise: Optional[str] = None
    world_overview: Optional[str] = None
    story_core: Optional[dict] = None
    status: Optional[str] = None
    target_words: Optional[int] = None
    cover_url: Optional[str] = None
    extra: Optional[dict] = None


class ProjectOut(BaseModel):
    id: uuid.UUID
    title: str
    genre: Optional[str]
    logline: Optional[str]
    premise: Optional[str]
    world_overview: Optional[str]
    story_core: Optional[dict]
    status: str
    target_words: Optional[int]
    cover_url: Optional[str]
    extra: Optional[dict] = None
    created_at: datetime
    updated_at: Optional[datetime]

    class Config:
        from_attributes = True
